internal/service/appstream/stack_user_association.go

```diff
@@ -0,0 +1,177 @@
+package appstream
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/appstream"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+)
+
+func ResourceStackUserAssociation() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceStackUserAssociationCreate,
+		ReadWithoutTimeout:   resourceStackUserAssociationRead,
+		DeleteWithoutTimeout: resourceStackUserAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"authentication_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(appstream.AuthenticationType_Values(), false),
+			},
+			"send_email_notification": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
```

Comment:
`send_email_notification` shouldn't be `ForceNew`, since the option only applies at creation-time; in all other cases, changed values should be ignored.